app/models/direct_file_data.rb

```diff
@@ -502,14 +504,33 @@ def fed_w2_state=(value)
     write_df_xml_value(__method__, value)
   end
 
+  def primary_blind
+    create_or_destroy_df_xml_node(__method__, true, 'VirtualCurAcquiredDurTYInd')
+    write_df_xml_value(__method__, "X")
+  end
+
+  def spouse_blind
+    create_or_destroy_df_xml_node(__method__, true, 'VirtualCurAcquiredDurTYInd')
+    write_df_xml_value(__method__, "X")
+  end
+
+  def is_primary_blind?
+    return false unless parsed_xml.at('PrimaryBlindInd')
+    true
```

Comment:
[dust] can you explain this section? It feels like we have two functions here `primary_blind` and `is_primary_blind` based on two different pieces of data in the XML? Can you help me understand?

Reply:
`primary_blind` is used as a setter to modify the XML. We only use it in `factories` to set the trait to true. The name is `primary_blind` to match the convention of the other setters in this class.

`is_primary_blind?` is the method actually used to determine the blindness of the filer.

Reply:
@jachan why do they set different values? As in, if `is_primary_blind` is reading the value of `'PrimaryBlindInd'` in the XML, I would expect that this is the value to set if we want to indicate they are blind. Why does the trait setter use `'VirtualCurAcquiredDurTYInd'` - where does that come from?

Reply:
The `'VirtualCurAcquiredDurTYInd'` arg in the `create_or_destroy_df_xml_node` method is the `after` arg. See https://github.com/codeforamerica/vita-min/blob/f3c0c353190a556f35c76c3ed7592a91a2b4af79/app/helpers/df_xml_crud_methods.rb#L6. 

The `after` arg specifies what the preceding node should be. In this case, we want to insert the `primary_blind` node (with node name fetched from the `selectors` dict in line 69 of this class) after the `VirtualCurAcquiredDurTYInd` node.

Reply:
@jachan ooh thanks for explaining. Huh! That is really unintuitive. Not your fault of course. I assume we can trust that the order of nodes doesn't change here?

Reply:
Yeah! It is weird. Asked CfA about how they construct the XML samples they use, and they said that their PMs get it from the IRS DF folks, since there isn't a public schema for the DF data. Existing DF samples with `<PrimaryBlindInd>` always have this field after `<VirtualCurAcquiredDurTYInd>` so I put it there.

I don't think the order matters necessarily, just the hierarchy of where this node is placed. And it's important to make sure that the sibling node `<VirtualCurAcquiredDurTYInd>` always exists in the base XML document. @tahsinaislam can you speak to how we can guarantee that this insert doesn't ever fail because `<VirtualCurAcquiredDurTYInd>` doesn't exist?

Reply:
Yep, the `PrimaryBlindInd` element in the XML only shows up if the applicant is blind. Otherwise it's missing and `write_df_xml_value` fails (I believe). A similar pattern with the `create_or_destroy_df_xml_node` method is used to set spouse information, which might also be missing (if the filer is single, for example). 

https://github.com/codeforamerica/vita-min/blob/main/app/models/direct_file_data.rb#L120

Where we differ from the existing pattern is that this value is boolean. In `mailing_street`, for example, we can just write the value, if `primary_blind` should be set, then we'll need to set it to `X` to indicate that this value is true. There's no real point in accepting a "value" here.:

```
  def mailing_street=(value)
    write_df_xml_value(__method__, value)
  end
```

That's my current understanding of how this works based on my conversation with Tahsina a month or so ago. Happy to talk more about this and refactor if needed!